### DISCUSSION TITLE
tab dots

### DISCUSSION THREAD

**Discussion by notwidow** [helix-editor/helix#3887] [Q&A]
Hi there, sorry for raising this issue but can you tell me how can i enable render for tabs only? in 
```ruby
[editor.whitesapce.render]
space = 'all' # this puts dots everywhere instead of spaces and i want it for tabs only
tab = 'all'
newline = 'none'
```
when i make space = 'none' then all render dots disappear. so how can i enable it only for tabs and not for spaces? Thank
Note: i also tried 
```ruby 
[editor.whitespace.render]
space = 'none'
tab = 'all'
tabpad = 'all'
newline = 'none'
```

**Comment by Omnikar** [comment]
It sounds like the file you're editing uses spaces for indentation (not tabs). Unfortunately, there is currently no way to configure the editor to only render indentation spaces and not other spaces.

**Reply by notwidow** [reply]
> It sounds like the file you're editing uses spaces for indentation (not tabs). Unfortunately, there is currently no way to configure the editor to only render indentation spaces and not other spaces.

yeah i want to migrate from vim. and yes those files are written in vim... where you know we have the option for that spaces with tabs.

**Comment by gabydd** [comment]
a table like this will render dots only where there are tabs
```toml
[editor.whitespace.render]
space = "none"
tab = "all"
newline = "none"

[editor.whitespace.characters]
tab = "·"
tabpad = "·"
```
 it also seems you have to close and open the editor to see them being rendered

**Reply by notwidow** [reply]
> a table like this will render dots only where there are tabs
> 
> ```toml
> [editor.whitespace.render]
> space = "none"
> tab = "all"
> newline = "none"
> 
> [editor.whitespace.characters]
> tab = "·"
> tabpad = "·"
> ```
> 
> it also seems you have to close and open the editor to seem them being rendered
  ```ruby                                                                                                              
    11 [editor.whitespace.render]                                                                                       
    10 space·=·'all'                                                                                                    
     9 tab·=·'all'                                                                                                      
     8 newline·=·'none'                                                                                                 
     7                                                                                                                  
     6 [editor.whitespace.characters]                                                                                   
     5 space·=·"·"                                                                                                      
     4 nbsp·=·"⍽"                                                                                                       
     3 tab·=·"·"                                                                                                        
     2 newline·=·"⏎"                                                                                                    
     1 tabpad·=·"·"   
```
if i space = 'none' in editor.whitespace.render then it show none...
